### PR TITLE
Korath Exiles Stage II - Firestorm Edit + Haven/Efreti Implementation

### PR DESCRIPTION
**Pull request by Saugia**
Edits the Firestorm sound and effect to more appropriately match.

Adds the Korath Haven (thumbnail to come), as well as a new "korath havens" .txt that contains the fleet + missions that spawn the Haven in Exile systems.

Replaces Efreti KWS's with new Efreti models, and adds an uncommon mining fleet of the smallest ship.

### CODE REVIEW

File: data/korath/korath ships.txt [hunk 1]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+ship "Korath Haven"
+	sprite "ship/korath haven"
+	"uncapturable"
+	attributes
+		category "Heavy Freighter"
+		cost 27690000
+		shields 196420
+		hull 320690
+		"required crew" 1940
+		"bunks" 10920
```
Comment (ravenshining) [inline data/korath/korath ships.txt]: If it's uncapturable we don't need to worry about the calculation, right?  This ought to be a few times larger.

```suggestion
		"bunks" 31920
```

File: data/korath/korath.txt [hunk 2]
```diff
@@ -378,7 +378,6 @@ fleet "Korath Civilian Large"
 	variant 1
 		"Korath Chaser (World-Ship)"
 
-
```
Comment (ravenshining) [inline data/korath/korath.txt]: Why are you eating the whitespace between factions?  If this really is an issue, suggest it in stage I.  Where I'll still argue against it but at least it'll be in the right place.

File: data/korath/korath ships.txt [hunk 3]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+		"mass" 12000
+		"drag" 84
+		"heat dissipation" .22
+		"fuel capacity" 1700
```
Comment (ravenshining) [inline data/korath/korath ships.txt]: Even though it's a superheavy that uses fuel weapons, I don't think it needs this much.  That's what Fuel Processors are for.

```suggestion
		"fuel capacity" 1000
```

File: data/korath/korath ships.txt [hunk 4]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+		"Fuel Processor" 7
+		"Korath Repeater Rifle" 1150
+		
+		"Reverser (Stellar Class)"
+		"Reverser (Planetary Class)"
+		"Thruster (Stellar Class)"
+		"Thruster (Planetary Class)"
+		"Steering (Stellar Class)"
+		"Steering (Planetary Class)"
+		"Hyperdrive"
```
Comment (ravenshining) [inline data/korath/korath ships.txt]:
```suggestion
		"Afterburner (Stellar Class)"
		"Reverser (Stellar Class)"
		"Thruster (Stellar Class)"
		"Thruster (Planetary Class)"
		"Steering (Stellar Class)"
		"Steering (Planetary Class)"
		"Scram Drive"
```

File: data/korath/korath ships.txt [hunk 5]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+	bay "Fighter" 179.5 80
+		angle 315
+		back
+	explode "tiny explosion" 30
+	explode "small explosion" 45
+	explode "medium explosion" 50
+	explode "large explosion" 40
+	explode "huge explosion" 50
+	"final explode" "final explosion large"
+	description "Korath Havens are massive nomadic habitats that hold all that is left of their people."
```
Comment (ravenshining) [inline data/korath/korath ships.txt]: I realise fixing descs is like my job but I'm so tired I don't want to lose track

```suggestion
	description "Korath Havens are massive nomadic habitats that hold much of their remaining population."
```

File: data/korath/korath ships.txt [hunk 6]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+
```
Comment (ravenshining) [inline data/korath/korath ships.txt:1242]:
```suggestion


```

File: data/korath/korath.txt [hunk 7]
```diff
@@ -378,7 +378,6 @@ fleet "Korath Civilian Large"
 	variant 1
 		"Korath Chaser (World-Ship)"
```
Comment (ravenshining) [inline data/korath/korath.txt]:
```suggestion


```

File: data/korath/korath.txt [hunk 8]
```diff
@@ -393,7 +392,6 @@ fleet "Kor Efret Home"
 	variant 2
 		"Korath World-Ship C"
```
Comment (ravenshining) [inline data/korath/korath.txt:407]:
```suggestion


```

File: data/korath/korath ships.txt [hunk 9]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+		"Firelight Storage Rack" 4
+		"Firelight Missile" 108
+		"Firestorm Torpedo" 62
+		"Firestorm Turret" 1
+
+		"Triple Plasma Core" 3
+		"Systems Core (Large)" 2
+		"Liquid Sodium Cooler" 2
+		"Fuel Processor" 7
+		"Korath Repeater Rifle" 1150
```
Comment (ravenshining) [inline data/korath/korath ships.txt]:
```suggestion
		"Korath Repeater Rifle" 1954
		"Microbot Defense Station" 214
```

File: data/korath/korath ships.txt [hunk 10]
```diff
@@ -851,6 +851,213 @@ ship "Korath World-Ship" "Korath World-Ship C"
...
+		"required crew" 1940
+		"bunks" 10920
+		"mass" 12000
+		"drag" 84
+		"heat dissipation" .22
+		"fuel capacity" 1700
+		"cargo space" 1864
+		"outfit space" 2630
+		"weapon capacity" 1181
+		"engine capacity" 516
```
Comment (ravenshining) [inline data/korath/korath ships.txt]: Probably need to raise outfit and engines for edits below but I'm to tired for maths

**Review by ravenshining — COMMENTED** [review]
Thank you so much!
I'd do all the things I suggested myself but I'm still very very tired from my trip and don't trust myself to remember otherwise

**Comment by Saugia** [comment]
I fixed the space, but I won't commit it yet. I'm working on rendering the thumbnail now, as well as the Efreti ships. I'll add those on to here and then let you know when it's ready.

**Comment by Saugia** [comment]
Alrighty- after I fix any remaining errors, it should be good to go! The Efreti ships are basically the same stats as their counterparts currently, though, aside from mass and drag. We can take a look at those later (as well as ship/image names).

**Pull request merged**